File: frontend/src/modules/estudiante/services/estudianteService.js
import http from '../../../shared/services/http';
import { paramsListado } from '../../../shared/utils/paginacion';

const estudianteService = {
  listar: (filtros = {}) => {
    const { estadoAptitud, ...rest } = filtros;
    const params = paramsListado({
      ...rest,
      aptitud: estadoAptitud ?? rest.aptitud,
    });
    return http.get('/estudiantes', { params });
  },
  obtenerPorId:    (id)       => http.get(`/estudiantes/${id}`),
  registrar:       (dto)      => http.post('/estudiantes', JSON.stringify(dto), { headers: { 'Content-Type': 'application/json' } }),
  editar:          (id, dto)  => http.put(`/estudiantes/${id}`, dto),
  eliminar:        (id)       => http.delete(`/estudiantes/${id}`),
  marcarApto:      (id, numeroPractica) =>
    http.patch(`/estudiantes/${id}/aptitud`, {
      aptitud: 'APTO',
      ...(numeroPractica != null ? { numeroPractica } : {}),
    }),
  marcarNoApto:    (id, motivo) => http.patch(`/estudiantes/${id}/aptitud`, { aptitud: 'NO_APTO', motivo }),
  obtenerExpediente: (id)     => http.get(`/expedientes/${id}`),
  importarExcel:   (archivo)  => {
    const form = new FormData();
    form.append('archivo', archivo);
    return http.post('/estudiantes/importar', form,
      { headers: { 'Content-Type': 'multipart/form-data' } });
  },
  subirDocumento:  (id, formData) => {
    return http.post(`/estudiantes/${id}/documentos`, formData,
      { headers: { 'Content-Type': 'multipart/form-data' } });
  }
};

export default estudianteService;

File: src/main/java/com/avh/practicas/estudiante/service/EstudianteService.java
package com.avh.practicas.estudiante.service;

import com.avh.practicas.estudiante.dto.EstudianteDto;
import com.avh.practicas.estudiante.entity.EstadoAptitud;
import com.avh.practicas.estudiante.entity.Estudiante;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface EstudianteService {
    Estudiante registrar(EstudianteDto dto);
    Estudiante editar(Long id, EstudianteDto dto);
    Estudiante marcarApto(Long id);
    Estudiante marcarApto(Long id, Integer numeroPractica);
    Estudiante marcarNoApto(Long id, String motivo);
    Optional<Estudiante> obtenerPorId(Long id);
    Optional<Estudiante> obtenerPorIdentificacion(String identificacion);
    Page<Estudiante> listar(String programa, String facultad, EstadoAptitud aptitud, String estadoPractica, String busqueda, Pageable pageable);
    void importar(List<Estudiante> estudiantes);
    Estudiante guardar(Estudiante estudiante);
    void eliminar(Long id);
}
